tests/dataimportcron_test.go

```diff
@@ -134,7 +134,16 @@ var _ = Describe("DataImportCron", func() {
 				},
 			}
 			snapshot = utils.WaitSnapshotReady(f.CrClient, snapshot)
-			deleted, err := utils.WaitPVCDeleted(f.K8sClient, name, ns, 2*time.Minute)
+			deleted, err := utils.WaitPVCDeleted(f.K8sClient, name, ns, 30*time.Second)
+			if err != nil {
+				// work around https://github.com/kubernetes-csi/external-snapshotter/issues/957
+				// it does converge after the resync period of snapshot controller (15mins)
+				cc.AddAnnotation(snapshot, "workaround", "triggersync")
+				err = f.CrClient.Update(context.TODO(), snapshot)
```

Comment:
So you are essentially just re-triggering a reconcile so it cleans up, what happens if that gets the same error, maybe retry a few times instead of just once? I don't know if that is overkill or not.

Reply:
Yeah good call I am not sure. It fails on
```
E1109 05:15:43.755196       1 snapshot_controller.go:191] error check and remove PVC finalizer for snapshot [datasource-test-63c30e708034]: snapshot controller failed to update datasource-test-63c30e708034 on API server: Operation cannot be fulfilled on persistentvolumeclaims "datasource-test-63c30e708034": the object has been modified; please apply your changes to the latest version and try again
```

Reply:
I think we won't need to retry more since 30 seconds have passed so it had plenty of times to catch up